datalad/support/gitrepo.py

```diff
@@ -2319,28 +2320,87 @@ def get_deleted_files(self):
+    def get_gitattributes(self, path, index_only=False):
+        """Query gitattributes for one or more paths
 
         Parameters
         ----------
-        all_: bool
-          Adds --all to git check-attr call
+        path: path or list
+          Path(s) to query. Paths may be relative or absolute.
+        index_only: bool
+          Flag whether to consider only gitattribute setting that are reflected
+          in the repository index, not just in the work tree content.
 
         Returns
         -------
         dict:
-          attribute: value pairs
+          Each key is a queried path (always relative to the repostiory root),
+          each value is a dictionary with attribute
+          name and value items. Attribute values are either True or False,
+          for set and unset attributes, or are the literal attribute value.
         """
-        out, err = self._git_custom_command(["."], ["git", "check-attr", "--all"])
-        assert not err, "no stderr output is expected"
-        out_split = [
-            # splitting by : would leave leading space(s)
-            [e.lstrip(' ') for e in l.split(':', 2)]
-            for l in out.split('\n') if l
-        ]
-        assert all(o[0] == '.' for o in out_split)  # for paranoid
-        return dict(o[1:] for o in out_split)
+        path = assure_list(path)
+        cmd = ["git", "check-attr", "-z", "--all"]
+        if index_only:
+            cmd.append('--cached')
+        stdout, stderr = self._git_custom_command(path, cmd)
+        # make sure we have one entry for each query path to
+        # simplify work with the result
+        attributes = {_normalize_path(self.path, p): {} for p in path}
+        attr = []
+        for item in stdout.split('\0'):
+            attr.append(item)
+            if len(attr) < 3:
+                continue
+            # we have a full record
+            p, name, value = attr
+            attrs = attributes[p]
+            attrs[name] = \
+                True if value == 'set' else False if value == 'unset' else value
+            # done, reset item
+            attr = []
+        return attributes
+
+    def set_gitattributes(self, attrs, attrfile='.gitattributes'):
+        """Set gitattributes
+
+        Parameters
+        ----------
+        attrs : list
+          Each item is a 2-tuple, where the first element is a path pattern,
+          and the second element is a dictionary with attribute key/value
+          pairs. The attribute dictionary must use the same semantics as those
+          returned by `get_gitattributes()`. Path patterns can use absolute paths,
+          in which case they will be normalized relative to the directory
+          that contains the target .gitattributes file (see `attrfile`).
+        attrfile: path
+          Path relative to the repository root of the .gitattributes file the
+          attributes shall be set in.
```

Comment:
> a use-case where having it as an option would come handy already with git?

Any .gitattributes file in a subdirectory?  We could assume the base name .gitattributes and change this argument to take a directory, though the current approach seems straightforward enough.

We don't use it at the moment, but there is also $GITDIR/info/attributes, though the relative to the repo root isn't convenient for that use case since $GITDIR isn't necessarily "root/.git".

Reply:
True, it isn't necessarily always `.gitattributes`, hence I went this way.

Reply:
Thanks